compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIERemoveMemorySpace.cpp

```diff
@@ -0,0 +1,145 @@
+/// the non-quant variants.
+class GenericTypeConvert : public ConversionPattern {
+ public:
+  GenericTypeConvert(MLIRContext *context, TypeConverter &converter)
+      : ConversionPattern(converter, MatchAnyOpTypeTag(), 0, context) {}
+  LogicalResult matchAndRewrite(
+      Operation *op, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const override {
+    llvm::SmallVector<Type> newResults;
+    if (isa<FunctionOpInterface>(op)) {
+      return failure();
+    }
+
+    (void)getTypeConverter()->convertTypes(op->getResultTypes(), newResults);
+    OperationState state(op->getLoc(), op->getName().getStringRef(), operands,
+                         newResults, op->getAttrs(), op->getSuccessors());
+    for (Region &r : op->getRegions()) {
+      Region *newRegion = state.addRegion();
+      rewriter.inlineRegionBefore(r, *newRegion, newRegion->begin());
+      TypeConverter::SignatureConversion result(newRegion->getNumArguments());
+      (void)getTypeConverter()->convertSignatureArgs(
+          newRegion->getArgumentTypes(), result);
+      rewriter.applySignatureConversion(&newRegion->front(), result);
+    }
+    Operation *newOp = rewriter.create(state);
+    rewriter.replaceOp(op, newOp->getResults());
+    return success();
+  }
+};
+
+class AMDAIERemoveMemorySpacePass
+    : public impl::AMDAIERemoveMemorySpaceBase<AMDAIERemoveMemorySpacePass> {
+  static bool isIllegalType(Type type) {
+    return getMemorySpaceScrubbedType(type) != type;
+  }
+
+ public:
+  void getDependentDialects(DialectRegistry &registry) const override {
+    registry.insert<memref::MemRefDialect, xilinx::AIE::AIEDialect,
+                    AMDAIE::AMDAIEDialect>();
+  }
+
+  void runOnOperation() override {
+    MemspaceTypeConverter converter;
+    ConversionTarget target(getContext());
+
+    // Operations are legal if they don't contain any illegal type.
+    target.markUnknownOpDynamicallyLegal([](Operation *op) {
+      if (auto funcOp = dyn_cast<FunctionOpInterface>(op)) {
+        for (Type type : funcOp.getArgumentTypes()) {
+          if (isIllegalType(type)) return false;
+        }
+        for (Type type : funcOp.getResultTypes()) {
+          if (isIllegalType(type)) return false;
+        }
+      }
+      for (Type type : op->getResultTypes()) {
+        if (type && isIllegalType(type)) return false;
+      }
+      for (Type type : op->getOperandTypes()) {
+        if (type && isIllegalType(type)) return false;
+      }
+      return true;
+    });
+
+    auto *ctx = &getContext();
+
+    RewritePatternSet patterns(&getContext());
+    patterns.insert<GenericTypeConvert>(ctx, converter);
+    populateAnyFunctionOpInterfaceTypeConversionPattern(patterns, converter);
+
+    if (failed(
+            applyFullConversion(getOperation(), target, std::move(patterns)))) {
+      signalPassFailure();
+    }
+
+    getOperation()->walk([&](xilinx::AIE::ObjectFifoCreateOp op) {
```

Comment:
Why is this needed for `ObjectFifoCreateOp`?

Reply:
I added a comment in the code answering this q.

Reply:
Ok, I am getting rid of this op, so soon we can just delete these lines. Not worth investigating a more 'sustainable' way of doing this at this point.